bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+    - if it has NOT received an `announcement_signatures_2` message:
+        - SHOULD retransmit the `announcement_signatures_2` message.
+
+A recipient node:
+- if the `short_channel_id` is NOT correct:
+    - SHOULD send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if the `partial_signature` is NOT valid as
+  per [Partial Signature Verification](#partial-signature-verification):
+    - MAY send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if it has sent AND received a valid `announcement_signatures_2` message:
+    - SHOULD queue the `channel_announcement_2` message for its peers.
+- if it has not sent `channel_ready`:
+    - MAY send a `warning` and close the connection, or send an `error` and fail
+      the channel.
+
+### Rationale
+
+The message contains the necessary partial signature, by the sender, that the
+recipient will be able to combine with their own partial signature to construct
+the signature to put in the `channel_announcement_v2` message. Unlike the legacy
+`announcement_signatures` message, `announcement_signatures_v2` only has one
+signature field. This field is a MuSig2 partial signature which is the
+aggregation of the two signatures that the sender would have created (one for
+`bitcoin_key_x` and another for `node_ID_x`).
+
+## The `channel_announcement_2` Message
+
+This gossip message contains ownership information regarding a taproot channel. 
+It ties each on-chain Bitcoin key that makes up the taproot output key to the 
+associated Lightning node key, and vice-versa. The channel is not practically 
+usable until at least one side has announced its fee levels and expiry, using 
+`channel_update_2`.
+
+See [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+for more information regarding the requirements for proving the existence of a
+channel.
+
+1. type: xxx (`channel_announcement_2`)
+2. data: (tlv_stream):
+   1. type: 0 (`announcement_sig`)
+   2. data:
+        * [`schnorr_signature`:`schnorr_signature`]
+   3. type: 1 (`features`)
+   4. data:
+        * [`...*byte`: `features`]
+   5. type: 2 (`chain_hash`)
+   6. data:
+        * [`chain_hash`:`chain_hash`]
+   7. type: 3 (`short_channel_id`)
+   8. data:
+        * [`short_channel_id`:`short_channel_id`]
+   9. type: 4 (`node_id_1`)
+   10. data:
+        * [`point`:`point`]
+   11. type: 5 (`node_id_2`)
+   12. data:
+       * [`point`:`point`]
+   13. type: 6 (`bitcoin_key_1`)
+   14. data:
+       * [`point`:`point`]
+   15. type: 7 (`bitcoin_key_2`)
+   16. data:
+       * [`point`:`point`]
+
+### Requirements
+
+The origin node:
+
+- MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that the channel was opened within:
+    - for the _Bitcoin blockchain_:
+        - MUST set `chain_hash` value (encoded in hex) equal
+          to `6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000`.
```

Comment:
I suggest the same trick we use for offers: omitting this field means it is 6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000.  And you must omit the field if it would be 6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000 (remember: both sides need to agree on exactly what this message looks like!).

Reply:
Note also: I *like* the use of the TLV here.  Because if we want to later, we can adapt this for pre-taproot channels, which would require more / different sigs.  Though maybe they'll all close by the time v1-apocolypse is here?

Reply:
responded [here](https://github.com/lightning/bolts/pull/1059/files#r1143366328)